File: python/stdlib/default/pass.py
try:
    from ...stdlib import Node as base
except ValueError:
    from stdlib import Node as base


class Node(base.Node):
    def __init__(self, verbose, args):
        super(Node, self).__init__("Pass", "stdlib.default.pass", {},
                                   {"in": "Object"},
                                   {"out": "Object"},
                                   "Pass the input to out.", verbose)
        self.args = args

    def tick(self, value):
        result = {}
        tag = None
        if "tags" in value and "in" in value["tags"]:
            tag = value["tags"]["in"]
        
        result = value["in"]
        
        if tag:
            return {"out": result, "tags":{"out":tag}}
        else:
            return {"out": result}

File: lua/stdlib/default/pass.lua
local myNode = {}
function myNode.init(node)
  node.tick = function(value)
    result = {}
    result["out"] = value["in"]
    return result
  end
end

function myNode.spec(node)
  node.name = "Pass"
  node.inputs["in"] = "Object"
  node.outputs["out"] = "Object"
  node.desc = "Pass an object. Does nothing."
end

return myNode
